cmake/kconfig.cmake

```diff
@@ -49,12 +49,17 @@ foreach(kconfig_target ${kconfig_target_list})
     )
 endforeach()
 
+# Bring in extra configuration files dropped in by the user or anyone else;
+# make sure they are set at the end so we can override any other setting
+file(GLOB config_files ${APPLICATION_BINARY_DIR}/*.conf)
+list(SORT config_files)
 set(
   merge_config_files
   ${BOARD_DEFCONFIG}
   ${CONF_FILE_AS_LIST}
   ${OVERLAY_CONFIG}
-  )
+  ${config_files}
```

Comment:
why not use OVERLAY_CONFIG instead? you just need to point to a 1 single file instead of polluting the output directory with *.conf files in this case. Even if you put it in the output, you can still reference it with OVERLAY_CONFIG

Reply:
Rationale in the commit message